### PR TITLE
SelectBox: Set "role" and "aria-labelledby" when fieldTemplate is rendered (T1230635)

### PR DESCRIPTION
**Pull request by jdvictoria**

### CODE REVIEW

**Comment by jdvictoria** [comment]
[SelectBox - The input element has incorrect role attribute value if async templates are used [T1230635]](https://trello.com/c/Zobx2cfX/8812-selectbox-the-input-element-has-incorrect-role-attribute-value-if-async-templates-are-used-t1230635)

File: packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js [hunk 1]
```diff
@@ -291,10 +292,15 @@ const DropDownEditor = TextBox.inherit({
         return 'none';
     },
 
+    _getRole() {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js]:
```suggestion
    _getAriaRole() {
```

File: packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js [hunk 2]
```diff
@@ -322,6 +328,10 @@ const DropDownEditor = TextBox.inherit({
         this.callBase();
     },
 
+    _setFieldAriaLabel() {
+        this.setAria('labelledby', TEMPLATE_CONTENT_LABEL);
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js]: The labelledby attribute should be linked with the ID of the label. Please check Artem's [sample](https://stackblitz.com/edit/vitejs-vite-cw79as?file=package.json) and see how it should work on SelectBox without fieldTemplate.

File: packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js [hunk 3]
```diff
@@ -322,6 +327,12 @@ const DropDownEditor = TextBox.inherit({
         this.callBase();
     },
 
+    _setFieldAriaLabel() {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js]: I think that we can use ['_setFieldAria'](https://github.com/EugeniyKiyashko/DevExtreme/blob/9fa928b2af16c227295fac7459aa2365ef6b7d6e/js/ui/text_box/ui.text_editor.base.js#L488) method intead. 
Also I suggest test the following scenarios: 
- aria-label is declared using 'inputAttr' option
- change label, labelMode at runtime, `aria-labelledby` should be removed if label was deleted and vise versa

File: packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js [hunk 4]
```diff
@@ -291,10 +291,15 @@ const DropDownEditor = TextBox.inherit({
         return 'none';
     },
 
+    _getAriaRole() {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js:294]: You can use this method for getting value 'combobox' in this line: https://github.com/EugeniyKiyashko/DevExtreme/blob/9fa928b2af16c227295fac7459aa2365ef6b7d6e/js/ui/drop_down_editor/ui.drop_down_editor.js#L253.

File: packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js [hunk 5]
```diff
@@ -322,6 +327,12 @@ const DropDownEditor = TextBox.inherit({
         this.callBase();
     },
 
+    _setFieldAriaLabel() {
+        if(this._label) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js]: Why we should leave this condition?

File: packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js [hunk 6]
```diff
@@ -376,6 +381,7 @@ const DropDownEditor = TextBox.inherit({
 
                 this._integrateInput();
                 isFocused && eventsEngine.trigger($input, 'focus');
+                isFocused && this._setFieldAria();
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js:384]: Why we call _setFieldAria method when input is focused only?

File: packages/devextreme/js/ui/text_box/ui.text_editor.base.js [hunk 7]
```diff
@@ -497,11 +497,9 @@ const TextEditorBase = Editor.inherit({
         const ariaLabel = inputAttr?.['aria-label'];
         const labelId = this._label.getId();
 
-        const value = ariaLabel ? undefined : labelId;
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/text_box/ui.text_editor.base.js:500]: Why we should change implementation of this method?

**Pull request closed without merging**